src/sentry/api/permissions.py

```diff
@@ -75,11 +75,13 @@ class (that is not StaffPermission) require this mixin because staff does not gi
     https://www.python.org/download/releases/2.3/mro/ to learn more.
     """
 
-    def has_permission(self, request, *args, **kwargs):
-        return super().has_permission(request, *args, **kwargs) or is_active_staff(request)
+    def has_permission(self, request, *args, **kwargs) -> bool:
+        # Check for staff before calling super to avoid catching exceptions from super
+        return is_active_staff(request) or super().has_permission(request, *args, **kwargs)
```

Comment:
we want the exceptions to bubble up?

Reply:
yes we catch them later in `dispatch` in api/base

Reply:
but does it make sense to catch and log here, and simply return false?

Reply:
If we return `False` here the error msg ends up being a generic 403 and I think the raised exceptions sometimes have useful messages and codes